### PR TITLE
Add CodeCatalyst definitions

### PR DESCRIPTION
**Pull request by JadenSimon**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

## Checklist

- [ ] I have read the **README** document
- [ ] I have read the **CONTRIBUTING** document
- [ ] My code follows the code style of this project
- [ ] I have added tests to cover my changes
- [ ] All new and existing tests passed

## License
I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: telemetry/definitions/commonDefinitions.json [hunk 1]
```diff
@@ -215,7 +215,7 @@
             "name": "credentialSourceId",
             "type": "string",
             "description": "Where credentials are stored or retrieved from",
-            "allowedValues": ["sharedCredentials", "sdkStore", "ec2", "ecs", "envVars", "other"]
+            "allowedValues": ["sharedCredentials", "sdkStore", "ec2", "ecs", "envVars", "awsId", "other"]
```
Comment (JadenSimon) [inline telemetry/definitions/commonDefinitions.json:218]: Based off https://github.com/aws/aws-toolkit-visual-studio-staging/pull/1231

**Comment by JadenSimon** [comment]
Need to fix VS codegen. It can't handle underscores in the field names ?

**Review by justinmk3 — APPROVED** [review]

**Review by rli — APPROVED** [review]

**Review by awschristou — APPROVED** [review]
We've typically avoided using underscores in the type definitions (eg: `cloudWatchLogsPresentation`), but these values are in use now so its worth committing a version that represents the current state of the Toolkits.

It would be worth revisiting this to remove the underscores if possible. We could get code generation collisions if later on we define different types named like `foo_updateBar` and `foo_update_bar`

**Comment by JadenSimon** [comment]
> It would be worth revisiting this to remove the underscores if possible. We could get code generation collisions if later on we define different types named like foo_updateBar and foo_update_bar

I've been leaning towards using the underscore to delineate namespaces moving forward because we already do it for the metric definitions. The types and definitions are both top-level identifiers and so they should follow the same pattern.  

Migrating telemetry data is not fun so this is something we should figure out soon. I'll write up a doc and then we can go from there.

File: telemetry/definitions/commonDefinitions.json [hunk 2]
```diff
@@ -559,6 +560,40 @@
...
+            "description": "Locality of the Amazon CodeCatalyst update dev environment request (i.e., from the thin client or the local IDE instance)",
+            "allowedValues": [
+                "remote",
+                "local"
+            ]
+        },
+        {
+            "name": "userId",
+            "type": "string",
+            "description": "Opaque AWS ID identifier"
```
Comment (shruti0085) [inline telemetry/definitions/commonDefinitions.json:586]: AWS Builder ID?

**Review by shruti0085 — APPROVED** [review]

**Pull request merged**